src/Sylius/Bundle/CoreBundle/Resource/StateMachine/CompositeStateMachine.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\Resource\StateMachine;
+
+use Sylius\Abstraction\StateMachine\StateMachineInterface;
+use Sylius\Bundle\ResourceBundle\Controller\RequestConfiguration;
+use Sylius\Bundle\ResourceBundle\Controller\StateMachineInterface as ResourceStateMachineInterface;
+use Sylius\Component\Resource\Model\ResourceInterface;
+use Webmozart\Assert\Assert;
+
+final class CompositeStateMachine implements ResourceStateMachineInterface
```

Comment:
Since this class is just an override of parent package to replace dependency, imo `Composite` is not the name for this class as it doesn't compose several somethings of the same type into one and it's not based on the input, e.g. provided `RequestConfiguration` doesn't change selected SM.

```suggestion
final class StateMachine implements ResourceStateMachineInterface
```
```suggestion
final class AbstractionStateMachine implements ResourceStateMachineInterface
```

Reply:
But it operates on the composite state machine from the abstraction. I'll stick to this name.